StochasticBarrierFunctions/benchmarks/barrier_synthesis.jl

```diff
@@ -0,0 +1,286 @@
+using StochasticBarrierFunctions, LazySets
+using YAXArrays, NetCDF, YAML
+
+abstract type SystemType end
+    struct LINEAR <: SystemType end
+    struct POLYNOMIAL <: SystemType end
+    struct NONLINEAR <: SystemType end
+
+abstract type BarrierType end
+    struct SOS <: BarrierType end
+    struct PWC <: BarrierType end
+
+abstract type PWCType end
+    struct DUAL_ALG <: PWCType end
+    struct CEGS_ALG <: PWCType end
+    struct GD_ALG   <: PWCType end
+
+function get_system_type(system_type_str) :: SystemType
+    return system_type_str == "linear"     ? LINEAR() :
+           system_type_str == "polynomial" ? POLYNOMIAL() :
+           system_type_str == "nonlinear"  ? NONLINEAR() :
+           error("Unknown system type: $system_type_str")
+end
+
+function get_barrier_type(barrier_type_str) :: BarrierType
+    return barrier_type_str == "SOS" ? SOS() :
+           barrier_type_str == "PWC" ? PWC() :
+           error("Unknown barrier type: $barrier_type_str")
+end
+
+function get_pwc_optimization_type(optimization_type_str) :: PWCType
+    return optimization_type_str == "DUAL_ALG" ? DUAL_ALG() :
+           optimization_type_str == "CEGS_ALG" ? CEGS_ALG() :
+           optimization_type_str == "GD_ALG"   ? GD_ALG() :
+           error("Unknown optimization type: $optimization_type_str")
+end
+
+function barrier_synthesis(yaml_file::String)
+    # Load config file
+    config = YAML.load_file(yaml_file)
+
+    # Define optimization type and make call
+    system_type_instance  = get_system_type(config["system_flag"])
+    barrier_type_instance = get_barrier_type(config["barrier_settings"]["barrier_type"])
+    call_barrier_method(config, system_type_instance, barrier_type_instance)
+end
+
+function extract_system_parms(config, system_type_str::LINEAR)
+    dim, A, b, σ = config["dim"], hcat(config["A"]...), config["b"], config["σ"]
+    state_space = Hyperrectangle(low=config["state_space"]["low"], high=config["state_space"]["high"])
+    return dim, A, b, σ, state_space
+end
+
+function extract_system_parms(config, system_type_str::POLYNOMIAL)
+    dim, f, σ = config["dim"], config["f"], config["σ"]
```

Comment:
Is `f` not a string? While we do not have the appropriate system type already (see comment below), this system once we make it would take an `AbstractPolynomialLike` from `MultivariatePolynomials`.

Reply:
Yes, the `AdditiveGaussianPolySystem `struct still needs to be defined in `system.jl`. I will do that in a separate PR to not convolute this one. Once it is ready, I will be build a bridge that converts `f `from a string to a MultivartiaPolynomial